ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -798,6 +798,37 @@ public void getLogSegmentsNotInJournalTest() throws InterruptedException, IOExce
   }
 
   /**
+   * Tests the cases where tokens have to be revalidated on a call to
+   * {@link PersistentIndex#findEntriesSince(FindToken, long)}.
+   * @throws StoreException
+   */
+  @Test
+  public void tokenRevalidationTest() throws StoreException {
```

Comment:
Once this is merged, can you add a test to BlobCompactorTest or IndexTest as to check that hard delete tokens are reset if compactor has compated the entry which the hard delete token was referring to. Also other case too, if not for compaction, hard delete should proceed as usual even if there are segments which was compacted as future segments.